x-pack/plugins/security_solution/server/endpoint/services/agent/clients/crowdstrike/crowdstrike_agent_status_client.ts

```diff
@@ -0,0 +1,172 @@
+        },
+      },
+    })) as ActionTypeExecutorResult<CrowdstrikeGetAgentOnlineStatusResponse>;
+
+    return keyBy(agentStatusResponse.data?.resources, 'id');
+  }
+  async getAgentStatuses(agentIds: string[]): Promise<AgentStatusRecords> {
+    const esClient = this.options.esClient;
+    const metadataService = this.options.endpointService.getEndpointMetadataService();
+    const sortField = 'crowdstrike.host.last_seen';
+
+    const query = {
+      bool: {
+        must: [
+          {
+            bool: {
+              filter: [
+                {
+                  terms: {
+                    'crowdstrike.host.id': agentIds,
+                  },
+                },
+              ],
+            },
+          },
+        ],
+      },
+    };
+
+    try {
+      const [searchResponse, allPendingActions] = await Promise.all([
+        esClient.search(
+          {
+            index: CROWDSTRIKE_AGENT_INDEX_PATTERN,
+            from: 0,
+            size: 10000,
+            query,
+            collapse: {
+              // TODO: check if we should use crowdstrike.cid instead
+              field: 'crowdstrike.host.id',
+              inner_hits: {
+                name: 'most_recent',
+                size: 1,
+                sort: [
+                  {
+                    [sortField]: {
+                      order: 'desc',
+                    },
+                  },
+                ],
+              },
+            },
+            sort: [
+              {
+                [sortField]: {
+                  order: 'desc',
+                },
+              },
+            ],
+            _source: false,
+          },
+          { ignore: [404] }
+        ),
+
+        getPendingActionsSummary(esClient, metadataService, this.log, agentIds),
+      ]).catch(catchAndWrapError);
+
+      const mostRecentAgentInfosByAgentId = searchResponse?.hits?.hits?.reduce<
+        Record<string, RawCrowdstrikeInfo>
+      >((acc, hit) => {
+        // TODO TC: check if we should use crowdstrike.cid instead
+        if (hit.fields?.['crowdstrike.host.id'][0]) {
+          acc[hit.fields?.['crowdstrike.host.id'][0]] =
+            hit.inner_hits?.most_recent.hits.hits[0]._source;
+        }
+
+        return acc;
+      }, {});
+
+      const agentStatuses = await this.getAgentStatusFromConnectorAction(agentIds);
+
+      return agentIds.reduce<AgentStatusRecords>((acc, agentId) => {
+        const agentInfo = mostRecentAgentInfosByAgentId[agentId].crowdstrike;
+
+        const agentStatus = agentStatuses[agentId];
+        const pendingActions = allPendingActions.find(
+          (agentPendingActions) => agentPendingActions.agent_id === agentId
+        );
+
+        acc[agentId] = {
+          agentId,
+          agentType: this.agentType,
+          // TODO: check if we should use crowdstrike.cid instead
+          found: agentInfo?.host.id === agentId,
+          isolated: agentInfo?.status === CROWDSTRIKE_NETWORK_STATUS.CONTAINED,
```

Comment:
Yeah we do this on the frontend using the pendingActions and isolated status. It's [the same logic as we use for endpoint](https://github.com/ashokaditya/kibana/blob/6ea9eb1079cebd136ae1ca1a46fe741941610192/x-pack/plugins/security_solution/public/common/components/agents/agent_status/agent_response_action_status.tsx#L27) and [AgentStatus](https://github.com/elastic/kibana/blob/14d459f5816ba1531f284a19858967809d853dd8/x-pack/plugins/security_solution/public/common/components/agents/agent_status/agent_status.tsx#L89) uses it.

Reply:
Right. but for external agents we may need to set the `isolated` property differently, since someone could have isolated the host via the 3rd party system and not from Kibana

Reply:
Oh indeed. We need to consider that situation.

Reply:
Agree 👍